pygfx/renderers/wgpu/meshrender.py

```diff
@@ -363,8 +406,47 @@ def fragment_shader(self):
 
             // Lighting
             $$ if lighting
-                let world_pos = varyings.world_pos;
-                let lit_color = lighting_{{ lighting }}(is_front, varyings.world_pos, varyings.normal, varyings.light, varyings.view, albeido);
+                // give a default lighting if no light in the scene?
+                $$ if not has_ambient_light and num_dir_lights == 0 and num_point_lights == 0
+                    let light_color = vec3<f32>(1.0, 1.0, 1.0);
+                    // let light_dir = -normalize(ndc_to_world_pos(vec4<f32>(0.0, 0.0, 1.0, 1.0)));
+                    let light_dir = (u_stdinfo.cam_transform_inv * vec4<f32>(0.0, 0.0, 1.0, 0.0)).xyz;
+                    let lit_color = lighting_{{ lighting }}(is_front, varyings.world_pos, varyings.normal, light_color, light_dir, albeido);
```

Comment:
Looks like there still is a default camera?

Reply:
> Why? Wouldn't this result in a lot of duplicate code for all the remaining mesh stuff?

It's mainly because I have made great changes to the renderer of MeshPhongMaterial , so I'm afraid other materials will be affected. :)

I think `ShaderLib` can solve the problem of code duplication. My idea is to put the repeated shader codes in shaderlib, and different renderers can splice and combine their own rendering logic as needed.

> > MeshPhongMaterial added a Boolean attribute 'flat_sharding ', which has the same affect as the current MeshFlatMaterial. (So MeshPhongMaterial may no longer be needed)
> 
> It looks like this makes it more like ThreeJS. Was that the idea?

My main consideration is that many different materials may have "flat_shading" attribute, not only MeshPhongMaterial. Materials such as MeshNormalMaterial can also use "Flat rendering mode". In the future, there may be materials of other lighting model, including PBR material, etc. 
Therefore, it may be better to take it as an attribute instead of providing only one (based on MeshPhongMaterial) MeshFasdMaterial.

> > The default light is removed. When using MeshPhongMaterial, you need to add a light in the scene to illuminate the target.
> 
> Mmm ... I quite like the idea of a default light. I can see how it feels a wee bit like a hack, but without it all lit objects become invisible if there are no lights. @Korijn thoughts?

Adding default lights can be a bit odd in logic. If someone just wants to show and view objects, they can use materials that are not affected by light (maybe we can provide more types of materials that are not affected by light)

In addition, I noticed that in some advanced lighting tutorials, there are some lighting based on the surrounding environment or map (IBL). There are no lights in the scene at this time,  adding default lights is more strange.

> > Now, wireframe property is realized by setting the PrimitiveTopology to "line_list", instead of shader code.
> 
> The reason that line_list was not used is that GPU lines are 1px thin, making them nearly invisible on hidpi screens. With the current implementation, the line width of the wire can be set.

I searched the webgpu specification for a long time. I thought it would provide this API for setting linewidth, but I really didn't find it. :}